Userland/Applications/HexEditor/GoToOffsetDialog.cpp

```diff
@@ -73,8 +73,8 @@ int GoToOffsetDialog::calculate_new_offset(int input_offset)
         VERIFY_NOT_REACHED();
     }
 
-    if (new_offset > m_buffer_size)
-        new_offset = m_buffer_size;
+    if (new_offset > m_buffer_size - 1)
```

Comment:
Please avoid putting unrelated stuff in your PR, that's never going to help you get them merged earlier.

This problem is (unsurprisingly) quite recurrent, so let's try to factorize every modification made to the cursor into one function, so every single check (and bug :)) is located at the same place. But yeah, that looks like work for another PR.